### ISSUE TITLE
How to use  predict interface for savedmodel that inputs have 'coo_sparse' what is SparseTensor?

### ISSUE THREAD

**Issue by zxgm** [NVIDIA/tensorrt-inference-server#989]
My tensorflow savedmodel signature_graph is follow:

> {'serving_default': inputs {
>   key: "input0"
>   value {
>     name: "input_1:0"
>     dtype: DT_INT32
>     tensor_shape {
>       dim {
>         size: -1
>       }
>       dim {
>         size: 100
>       }
>     }
>   }
> }
> inputs {
>   key: "input1"
>   value {
>     name: "input_2:0"
>     dtype: DT_INT32
>     tensor_shape {
>       dim {
>         size: -1
>       }
>       dim {
>         size: 100
>       }
>     }
>   }
> }
> inputs {
>   key: "input2"
>   value {
>     dtype: DT_FLOAT
>     tensor_shape {
>       dim {
>         size: -1
>       }
>       dim {
>         size: -1
>       }
>     }
>     coo_sparse {
>       values_tensor_name: "input_3/values:0"
>       indices_tensor_name: "input_3/indices:0"
>       dense_shape_tensor_name: "input_3/shape:0"
>     }
>   }
> }
> inputs {
>   key: "input3"
>   value {
>     dtype: DT_FLOAT
>     tensor_shape {
>       dim {
>         size: -1
>       }
>       dim {
>         size: -1
>       }
>     }
>     coo_sparse {
>       values_tensor_name: "input_4/values:0"
>       indices_tensor_name: "input_4/indices:0"
>       dense_shape_tensor_name: "input_4/shape:0"
>     }
>   }
> }
> inputs {
>   key: "input4"
>   value {
>     name: "input_5:0"
>     dtype: DT_INT32
>     tensor_shape {
>       dim {
>         size: -1
>       }
>       dim {
>         size: 100
>       }
>     }
>   }
> }
> outputs {
>   key: "prediction"
>   value {
>     name: "dense_3/Sigmoid:0"
>     dtype: DT_FLOAT
>     tensor_shape {
>       dim {
>         size: -1
>       }
>       dim {
>         size: 80
>       }
>     }
>   }
> }
> method_name: "tensorflow/serving/predict"
> }

I have called status interface result is follow:

> id: "inference:0"
> version: "1.6.0"
> uptime_ns: 177063471915
> model_status {
>   key: "serving_model"
>   value {
>     config {
>       name: "serving_model"
>       platform: "tensorflow_savedmodel"
>       version_policy {
>         latest {
>           num_versions: 1
>         }
>       }
>       max_batch_size: 128
>       input {
>         name: "input4"
>         data_type: TYPE_INT32
>         dims: 100
>       }
>       input {
>         name: "input3"
>         data_type: TYPE_FP32
>         dims: -1
>       }
>       input {
>         name: "input2"
>         data_type: TYPE_FP32
>         dims: -1
>       }
>       input {
>         name: "input1"
>         data_type: TYPE_INT32
>         dims: 100
>       }
>       input {
>         name: "input0"
>         data_type: TYPE_INT32
>         dims: 100
>       }
>       output {
>         name: "prediction"
>         data_type: TYPE_FP32
>         dims: 80
>       }
>       instance_group {
>         name: "serving_model_0"
>         count: 1
>         gpus: 0
>         kind: KIND_GPU
>       }
>       default_model_filename: "model.savedmodel"
>     }
>     version_status {
>       key: 1
>       value {
>         ready_state: MODEL_READY
>       }
>     }
>   }
> }
> status_stats {
>   success {
>     count: 3
>     total_time_ns: 235139
>   }
> }
> ready_state: SERVER_READY
> 

but when i called predict, I hit error:

> {"error": "code: INTERNAL msg: \"Tensor :0, specified in either feed_devices or fetch_devices was not found in the Graph\" server_id: \"inference:0\" request_id: 24"}
I have no method to resolve the problem.Anyone can help me,thank you very much.

**Comment by GuanLuo** [comment]
Can you run the model in TensorFlow directly? I wonder if any special handling is required... If it can run directly, can you try to run with instance_group setting `kind: KIND_MODEL`?

**Comment by zxgm** [comment]
I try use the model in tensorflow directly that run normal,and the instance_group setting kind:KIND_MODEL that be used to set KIND_GPU or KIND_CPU.I have found a method to resolve the problem that first convert sparse tensor to dense tensor,then generate the model

**Comment by GuanLuo** [comment]
Can you share your code to run the model in TensorFlow directly? I wonder why it won't work with TRTIS then, because internally TRTIS also uses TensorFlow's API to execute the model.

**Comment by zxgm** [comment]
I am sorry to tell you,I have not completely source code.The part of code to create sparseTensor that follows:

>     cross_feature_num = 1000
>     cross_feature_indices = tf.keras.layers.Input(shape=(cross_feature_num),
>                                             dtype="float32", sparse=True)
but i use the follow code to replace over code,that is work.
  

>   user_feature_num = 1000
>     cross_feature_num = 1000
>     user_feature_input = tf.keras.layers.Input(shape=(user_feature_num,), dtype='int32')
>     cross_feature_indices = tf.keras.layers.Input(shape=(2,), dtype='int64')
>     cross_feature_values = tf.keras.layers.Input(shape=(), dtype='float32')
> 
>     cross_feature_input = \
>         tf.keras.layers.Lambda(lambda x: tf.SparseTensor(indices=x[0], values=x[1], dense_shape=[tf.shape(user_feature_input)[0], cross_feature_num]))\
>             ([cross_feature_indices, cross_feature_values])

The savedmodel created on tensorflow1.15,but i want to use in tensorflow1.14

**Comment by GuanLuo** [comment]
I see, so a sparse tensor is created differently from a normal tensor.
